pkg/internal/executors/executor_sbt.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022, Pulumi Corporation.  All rights reserved.
+
+package executors
+
+import (
+	"os"
+	"strings"
+
+	"github.com/pulumi/pulumi-java/pkg/internal/fsys"
+)
+
+type sbt struct{}
+
+var _ javaExecutorFactory = &sbt{}
+
+func (s sbt) NewJavaExecutor(opts JavaExecutorOptions) (*JavaExecutor, error) {
+	ok, err := s.isSbtProject(opts)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, nil
+	}
+	probePaths := []string{opts.UseExecutor}
+	if opts.UseExecutor == "" {
+		probePaths = []string{"./sbt", "sbt"}
+	}
+	cmd, err := fsys.LookPath(opts.WD, probePaths...)
+	if err != nil {
+		return nil, err
+	}
+	return s.newSbtExecutor(cmd)
+}
+
+func (sbt) isSbtProject(opts JavaExecutorOptions) (bool, error) {
+	if strings.Contains(opts.UseExecutor, "sbt") {
+		return true, nil
+	}
+	sbtMarkers := []string{
+		"project/build.properties",
+		"build.sbt",
+	}
+	for _, p := range sbtMarkers {
+		_, err := os.Stat(opts.WD.Path() + "/" + p)
```

Comment:
`+"/"+` will run into trouble on Windows, the traditional Go answer is filepath.Join that makes it cross-platform.

The other little nit is that the other executor code is written against `fsys` abstraction not `os` which unlocks unit-testing in memory (fsys can be populated by in-memory maps instead of touching the filesystem).

```
// Checks if a file exists.
func FileExists(dir fs.FS, path string) (bool, error) {
	_, err := fs.Stat(dir, path)
	if err != nil && os.IsNotExist(err) {
		return false, nil
	}
	return true, err
}
```
 

Here this would simply be:

```
if fsys.FileExists(opts.WD, p) {
   return true, nil
}
```

Reply:
I see. Thank you! :bow: